content/posts/2024-11-05-flang-new.md

```diff
@@ -0,0 +1,444 @@
+  contributing F18 to the LLVM Project.
+* **April 2019** F18 is [approved for migration](https://discourse.llvm.org/t/f18-is-accepted-as-part-of-llvm-project/51719)
+  into the LLVM Project monorepo.
+
+  At this point F18 was only the early parts of the compiler, it could not generate
+  code (later `fir-dev` work addressed this). Despite that, it moved into `flang`
+  in the monorepo, ready for the day it was complete.
+* **August 2019** The [first appearance](https://github.com/flang-compiler/f18/commit/b6c30284e7876f6ccd4bb024bd5f349128e99b7c)
+  of the `flang.sh` driver wrapper script, more on this later.
+* **December 2019** The [plan](https://discourse.llvm.org/t/flang-landing-in-the-monorepo/54022)
+  for rewriting the F18 git history to fit into the LLVM project is announced.
+  This effort was led by Arm, with Peter Waller (Arm) going so far as to write
+  a [custom tool](https://github.com/llvm/llvm-project/commit/55d5e6cbe2509a24132d056e1f361dc39312929b#diff-c389405236998090c7c8b9741506f01fb28abbd7da52e9566323c585ac0eb89cL910)
+  to rewrite the history of F18.
+* **Februrary 2020** The [plan](https://discourse.llvm.org/t/plan-for-landing-flang-in-monorepo/54546)
+  for improvements to F18 to meet the standards required for inclusion in the
+  LLVM monorepo is announced by Richard Barton (Arm).
+* **April 2020** Upstreaming of F18 into the LLVM monorepo is
+  [completed](https://github.com/llvm/llvm-project/commit/b98ad941a40c96c841bceb171725c925500fce6c).
+
+At this point what was in the LLVM monorepo was F18, the rewritten frontend of
+Classic Flang. Classic Flang remained unchanged, using the PGI based frontend.
+
+Around this time work started in the Classic Flang repo on the `fir-dev` branch
+that would enable code generation when using F18.
+
+So for the next events remember that this whole time Classic Flang was (and still
+is today) in use. Classic Flang's binary name was `flang`, just like the folder F18 now occupied
+in the LLVM monorepo.
+
+* **October 2020** Flang is included in an LLVM release for the first time in
+  LLVM 11.0.0. There was an `f18` binary and the aforementioned script
+  `flang.sh`.
+* **October 2021** First appearance of the `flang-new` binary, in the LLVM 13.0.0
+  release.
+* **April 2022** NVIDIA [ceases development](https://discourse.llvm.org/t/nvidia-transition-from-fir-dev/61947)
+  of the `fir-dev` branch in the Classic Flang project. Upstreaming of `fir-dev`
+  to the LLVM Project begins around this date.
+* **April 2022** `flang-new` can now do code generation if an experimental flag
+  is used. Enabled by the code upstreamed into the LLVM Project from `fir-dev`.
+* **May 2022** The `flang.sh` driver script becomes `flang-to-external-fc`. It
+  allows the user to use `flang-new` to process Fortran source code then send
+  the result to an existing Fortran compiler such as gfortran.
```

Comment:
"process" -> "parse and unparse" (if you want to be specific)

Reply:
Done.